db/migrations/20180404165800_assign_stacks_to_buildpacks.rb

```diff
@@ -0,0 +1,30 @@
+require 'yaml'
+
+def stacks_yml
+  stacks_yml_path = ENV.fetch('STACKS_YML', nil)
+  YAML.safe_load(File.read(stacks_yml_path)) if stacks_yml_path && File.exist?(stacks_yml_path)
+end
+
+def default_stack
+  stacks_yml['default'] if stacks_yml
```

Comment:
The result of `stacks_yml` should be cached, even by doing something like:
```
@stacks_yml ||= begin
  stacks_yml_path = ENV.fetch('STACKS_YML', nil)
  YAML.safe_load(File.read(stacks_yml_path)) if stacks_yml_path && File.exist?(stacks_yml_path)
end
```

Reply:
with @ericpromislow

Reply:
Can you help me understand what is the concern with reading a fairly small yaml file to get a single key a handful of times during a single migration?

Implementing your suggestion appears to introduce global state, breaking the tests. This is the diff of what I applied:

```
--- a/db/migrations/20180404165800_assign_stacks_to_buildpacks.rb
+++ b/db/migrations/20180404165800_assign_stacks_to_buildpacks.rb
@@ -1,8 +1,10 @@
 require 'yaml'
 
 def stacks_yml
-  stacks_yml_path = ENV.fetch('STACKS_YML', nil)
-  YAML.safe_load(File.read(stacks_yml_path)) if stacks_yml_path && File.exist?(stacks_yml_path)
+  @stacks_yml ||= begin
+    stacks_yml_path = ENV.fetch('STACKS_YML', nil)
+    YAML.safe_load(File.read(stacks_yml_path)) if stacks_yml_path && File.exist?(stacks_yml_path)
+  end
 end
 
 def default_stack
```

Reply:
Not a dealbreaker. We only use that pattern when you new up a instance of a class so that the yaml  isn't cached forever.